packages/host/app/components/prerendered-card-search.gts

```diff
@@ -0,0 +1,120 @@
+import { TemplateOnlyComponent } from '@ember/component/template-only';
+import { hash } from '@ember/helper';
+import { service } from '@ember/service';
+
+import { buildWaiter } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+
+import { tracked } from '@glimmer/tracking';
+
+import { WithBoundArgs } from '@glint/template';
+
+import { restartableTask } from 'ember-concurrency';
+
+import { flatMap } from 'lodash';
+
+import { PrerenderedCard, Query } from '@cardstack/runtime-common';
+
+import { Format } from 'https://cardstack.com/base/card-api';
+
+import CardService from '../services/card-service';
+
+import PrerenderedCardComponent from './prerendered';
+
+const waiter = buildWaiter('prerendered-card-search:waiter');
+
+interface ResultsSignature {
+  Element: undefined;
+  Args: {
+    instances: PrerenderedCard[];
+  };
+  Blocks: {
+    default: [
+      item: WithBoundArgs<typeof PrerenderedCardComponent, 'card'>,
+      cardId: string,
+      index: number,
+    ];
+  };
+}
+
+const ResultsComponent: TemplateOnlyComponent<ResultsSignature> = <template>
+  {{#each @instances as |instance i|}}
+    {{yield (component PrerenderedCardComponent card=instance) instance.url i}}
+  {{/each}}
+</template>;
+
+interface Signature {
+  Element: undefined;
+  Args: {
+    query: Query;
+    format: Format;
+    realms: string[];
+  };
+  Blocks: {
+    loading: [];
+    response: [
+      {
+        count: number;
+        Results: WithBoundArgs<typeof ResultsComponent, 'instances'>;
+      },
+    ];
```

Comment:
This is doing a lot of work to give callers *some* access to array methods, when it would be better to just give them an array.

```suggestion
    response: [cards: (ComponentLike<{ Args: {} }> & { id: string })[]];
```

Reply:
I did this because I don't know how to create an invokable instance of a component in Javascript. How is this done?

Reply:
(or is there a way to map over data in glimmer and use the component helper?)